### ISSUE TITLE
Encryption enabled drops to busy box shell

### ISSUE THREAD

**Issue by jon-bit** [pieroproietti/penguins-eggs-legacy#338]
I continue to get errors when I install Linux on my secondary laptop with eggs. I usually use disk encryption and I don't want to avoid it, however disk encryption may be unusable because every time I select disk encryption on the install, it crashes and doesn't work but if I DON'T use disk encryption, it does work. So I'm not going to mind about how to fix this, but I'm including a screenshot of the full error. Please let me know if you have any suggestions, thank you for the amazing project and for the help.
![_2024-02-29-22-45-47-704](https://github.com/pieroproietti/penguins-eggs/assets/59271817/e10d3d13-3889-4595-959d-ba28dff515fa)

**Comment by jon-bit** [comment]
UPDATE: I disabled c states and now it says this.
![2024-02-29-23-03-27-242](https://github.com/pieroproietti/penguins-eggs/assets/59271817/c94fc715-8ade-4c8f-b822-84d9ab28af17)

**Comment by pieroproietti** [comment]
Hi,
 I can try to make a test. I'm not so expert on encrypted machine, but i can prepare a VM with encripted disk and see that happen.

**Comment by jon-bit** [comment]
Thanks @pieroproietti . I appreciate it. I used mint with a number of apps just so you know in advance what my setup looked like. let me know how it goes.

**Comment by pieroproietti** [comment]
Hi @jon-bit 

I just made - just a test - using Arch -  created a VM with encryped root on /dev/mapper/root and create a new ISO from it with eggs.

Surprising work nicelly, like it was on a normal partition and the ISO is produced. 

I reinstalled the resulting ISO and get a normal system - not encrypted - but this is normal becouse I don't have an option in krill (the eggs CLI system installer) to select for an encrypted partition.

I don't know if calamares let to create encrypted partition, another possible test is that.

**Comment by pieroproietti** [comment]
just made a test with calamares on the same VM. 
I added my colibri configuration, just an XFCE plus tools to develop, and added calamares on the encripted VM, then make a new ISO.
The new ISO start and install, from calamares I see the possibility to create a luks2 volume, but it seem there is no way to use is to create a root partition. This I think depend on calamares - or from my ignorance of the argumento - more than other.

In short seem eggs is almost trasparent to encrypted or not encripted systems.

OK, I discovered the way to create encrypt partition with calamares, just a click on [] encrypt and put passwork and I'm nicelly installing the ISO, not problems at all using Arch and calamares 3.3.x. 

The resulting installation work happily.

**Comment by pieroproietti** [comment]
# Again  a note
I thought I was done with my answer, and I was moving on to take a look at eggs to see if I can improve anything -- I recently discovered Copilot and AI in general -- and I must say it helps.

So I went to the krill code, the installer, and realized that there is in fact an option in krill for encrypted partitions and tried it on the same VM as above.

The result was this.

![instaling-encrypted-](https://github.com/pieroproietti/penguins-eggs/assets/958613/4af27874-91e9-49ab-9367-b883a395c8a0)

It doesn't ask me for the password to open the encrypted partitions, I would have to dive into the code to figure out what changes to make.

The CLI installation code is all in [krill](https://github.com/pieroproietti/penguins-eggs/tree/master/src/krill)

Try to take a look at it if you can provide some suggestions.

if on the other hand, you can't or won't, try using calamares for reinstallation, in this one I used krill and evidently something is missing.

**Comment by jon-bit** [comment]
"calamares let to create encrypted partition"

Hi. Thanks for the update. In the disk part of the install their should be a "encrypt disk" button. That's what I used. I'll post a photo of where it is If you want me to. Also I did use calamares. I honest don't know what to do. I have a lot of packages so let me try less and I will get back to you.

**Comment by jon-bit** [comment]
UPDATE: I have some news. I did an arch install with the CLI and got the error @pieroproietti is getting BUT it is different when I do it on mint (my OS of choice). That one drops to busy box rather then emergency shell. I don't know what to do here but I plan on trying another OS latter this week. Let me know if anyone is able to help and thanks in advance.

**Comment by jon-bit** [comment]
I just wanted to give an update that on the last few updates to eggs. As of 9.6.40 it is still not working. Their is no rush but I saw on the last few updates LUKS updates and I'm not a coder so I don't know if this means their has been a attempt to fix this, but I just wanted to say if their has been it has not worked yet at least for me and my modified mint distro.

UPDATE: Also wanted to add that something strange is I have to enter my encryption password but as I have stated I use mint. Not arch. So I have no clue what is happening but I am exited for a update to fix this.

**Comment by jon-bit** [comment]
UPDATE: I don't know if this is fixed yet But new error Now it says
"ioremap error for" PCI device number (I think) " requested 0x2, got 0x0"

@pieroproietti I don't know what is different but do you have a idea? 

Also, thanks for the effort. This is an amazing project.

**Comment by pieroproietti** [comment]
Hi @jon-bit 
I added some things criptedclone mostly, but it's not related on your question. Actually Im on v9.7.3l and the change was mostly in using pemGUI as GUI for eggs. Due the fact I have again same troubles with arch, I didn't updated It.

Thanks for the compliments, help always a little, I hope to have much time in the future.

**Comment by jon-bit** [comment]
Wanted to give a quick update with 10.0.2 out. I don't know if any attempts to fix this problem has been made (witch is fine. I understand their could be a bigger priority) however, I have not yet seen a fix. Their is no rush, but just wondering if their is a ETA to a fix because I want to use my iso (because this project is so cool) but I always use encryption, and it has been about 3 months now. Just a quick question for @pieroproietti , any idea when this will be fixed? Again major respect for amazing software.

P.S. Any ETA for the Fedora builds too?

**Comment by jon-bit** [comment]
UPDATE: It may be fixed. At least I did a mint install this afternoon and it worked. I will try arch in a few hours. If it works I will close as solved.

**Comment by jon-bit** [comment]
UPDATE: Test it on arch and now it it says "INIT NOT FOUND". Don't know whats up with that but when I boot the is that is what I get. Don't know if I did something wrong on the install (archinstall command) but if it works for someone else let me know and I will close.

**Comment by jon-bit** [comment]
I don't know what but it's not working anymore. I can 100% confirm it was working on 10.0.3 but not I'm seeing other wise as of forward. @pieroproietti It eggs made so you can make an egg of and egg? Like you make a iso off of a eggs made system. That could very well be the problem.

**Comment by pieroproietti** [comment]
Of course you can, once you install an ISO or a remastered ISO with eggs, must to be the same. 

Now, I don't know your specific case - you are speacking about arch - arch is again at 10.0.3 version.

 https://aur.archlinux.org/packages/penguins-eggs

**Comment by jon-bit** [comment]
sorry I meant mint

**Comment by pieroproietti** [comment]
Do you are using Mint based on Ubuntu or on Debian? That version are you using?

**Comment by jon-bit** [comment]
Sorry I'm a bit confused on what you mean. If I understand this right, I'm saying I am using Linux mint cinnamon. 

https://linuxmint.com/download.php

**Comment by pieroproietti** [comment]
So you are using the last Linux Mint 21.3 "Virginia", based on Ubuntu focal... 

I have not an answer, unfortunatelly, just noted today who for Ubuntu bionic based distro, You can try:

live-boot package: on `bionic` - for same reason - when the system is installed, directory `/lib/live/boot` is erased. The system work, eggs work and can produce, but the resulting ISO will not boot! To solve this problem, before generate the ISO, give: `sudo apt install live-boot --reinstall`. This will restore `/lib/live/boot` and it's full contents.

**Comment by jon-bit** [comment]
I'll give that try in a bit. thank you. I'll update as soon as I try it.

**Comment by jon-bit** [comment]
Didn't work. I have no clue why it is just not working now. I know for a fact 10.0.3 did encryption.

**Comment by jon-bit** [comment]
hmmmmmmm. @pieroproietti I'm out of ideas but do you think the grub could be it? It looks like that to me. Give me a bit and I'll post a photo of my config.

**Comment by jon-bit** [comment]
After much trouble and rage trying to upload this (It actually took 45 min), here it is: 
![2024-06-20_202808](https://github.com/pieroproietti/penguins-eggs/assets/59271817/856c1029-eba1-4476-a15c-49e0e01f8719)

**Comment by jon-bit** [comment]
I think It was  working on 10.0.18? I don't know. It "worked" but the next version came along and same thing again.

**Comment by jon-bit** [comment]
HUGE UPDATE! 

After poking around for a answer I found that the user directory (when enabling encryption) is NOT being made. I enabled some things (Running in chroot: "dracut -f" and "modprobe ahci") and it did boot but =failed to start the GUI. I tried to get to the normal user but their was no home directory but the user did exist. But even stranger it DID exist in the chroot but not in the failed GUI. I will send 2 screenshots in a min. It's hard to explain. I can show better then tell.

**Comment by jon-bit** [comment]
Here they are. @pieroproietti any ideas?
![Screenshot from 2024-08-05 19-54-49](https://github.com/user-attachments/assets/ff6a6af0-e240-4133-8aac-998653315975)
![Screenshot from 2024-08-05 20-01-34](https://github.com/user-attachments/assets/bb68c661-779f-4dbd-b377-9817cda9b8cf)

**Comment by pieroproietti** [comment]
I see `bin.usr-is-merged` and  `sbin.user-is-merged` and, just now noted you are using dracut.

I saw the files or folders also in same occasions on my live, so I thought was mine, but just tryed to find their names on the code and I was not able to find.

You are on Linuxmint, as far as I know it don't have `dracut` but `mkinitramfs`, did you installed it? 

If bad or well it work on Linuxmint is interesting, I'm interested on dracut too, for fedora, who I'm trying to include on supported distros.

**Comment by jon-bit** [comment]
@pieroproietti Well I don't know what did it but now I'm guessing it was `modprobe ahci`. However I'm not familiar on bin.usr-is-merged and sbin.usr-is-merged. Do you know what is wrong?

**Comment by jon-bit** [comment]
@pieroproietti I think I might have a idea. How do you enable kernel mods on eggs? It's a long shot/long story but it might work.

**Comment by pieroproietti** [comment]
The question is: I never compiled a kernel! My fault... :-D

Yes, probably can work, but I'm fighting to introduce Alpine, fedora and SuSE on eggs... With Alpine I'm in a good point, there are real images working, but lacks again an APKBUILD to create package, work only on BIOS for now, will help not little the presence of an APKBUILD for calamares too, I'm installing with my krill. Krill need a review too for this distro on keyboard configuration, due it's different. 

On fedora and SuSE, I'm locked at dracut, and from the fact I prefere go where I can do more, so at the moment I'm pushing more Alpine.

So, a lot of things boiling actually. 

I hope first or late someother jump on, fork the project and start to send patchs. Materials to learn there is, the code can be read, I can answer, there is guide, a README who alreay is a manual and I wrote lately some others documentation: WAY-TO-ALPINE. WAY-TO-ARCH, etc.

Have a nice friday @jon-bit

**Comment by jon-bit** [comment]
Hi @pieroproietti. Thanks for the response. 2 things.

1. Do you have any idea of a costume kernel for the time being that might fix this?
2. I don't think you need to compile the kernel to enable the mods every time. Just run ```modprobe [name of mod]``` and it will enable.

I understand that you are working hard and I respect that, I'm not asking for this to be fixed right now, I'm just asking if eggs should do that if I enable them?

**Comment by pieroproietti** [comment]
eggs, just remaster the system, so is you have a custom kernel it will be
used, at the moment not less, nor more!

**Comment by jon-bit** [comment]
OK, I'll get back if I get an update. Thanks for the help.

**Comment by jon-bit** [comment]
As of 10.0.25 The CLI install works with encryption. 

I used Full encryption, small swap, and ext4 on the install disk. When looking at the GUI, I think it is the swap or boot partitions(can't say why however. I just see it does only makes one of them).
